### DISCUSSION TITLE
No Alerts on fresh install on NUC (tried multiple versions)

### DISCUSSION THREAD

**Discussion by s0meguy1** [Security-Onion-Solutions/securityonion#11813] [2.4] [answered]
### Version

2.4.30

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Eval

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

32

### Storage for /

1TB

### Storage for /nsm

627G

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

Hello,

I am not receiving any kind of alerts on a default setup. bond0 is seeing traffic, I have the default alert rules setup, and when I replay test data in the grid area, it fails to produce alerts. I have seen other complaints of this in the discussion area, however they are referred to the setup guide. I have followed it exactly with no luck. I've also tried installing the older versions of SO via ISO (2.4.20, 2.4.10) on bare metal, along with installing proxmox on the bare metal NUC and installing the ISO, installing Debian/Ubuntu on bare metal and installing SO - also with no luck. 2.3 worked great on this device and I wanted to upgrade to 2.4 because of the feature set, but I just cannot get it working. I don't know where to begin looking for an issue like this. I would appreciate some guidance. Here are almost all logs from /opt/so/log/:
https://drive.google.com/drive/folders/1jPV6VuN1abFbcFzHFvnl3BdLKI31uLD5

I've also attached some screenshots of when I replay test data in grid, along with an influxdb screenshot to show that it is seeing network traffic
![Screenshot from 2023-11-16 09-20-58](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/4cdcb168-af45-4cb4-8326-af948481dabf)
![Screenshot from 2023-11-16 09-20-42](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/ee2eae3a-a30d-43f6-bf44-f80cc678302f)
![Screenshot from 2023-11-16 09-20-29](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/2068a6a1-e0c5-4e95-9080-da472bdb17df)
![Screenshot from 2023-11-16 09-19-39](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/d7c168f5-3711-4ecf-b520-7689cc027a37)
![Screenshot from 2023-11-16 09-18-18](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/ec2cb92a-5031-4cc4-a1aa-77a114431e83)
![Screenshot from 2023-11-16 08-10-49](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/30f45008-2bd4-40ab-8b88-c36d8fcc69e3)

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by dougburks** [comment]
Have you tried following the Troubleshooting Alerts section in the documentation?
https://docs.securityonion.net/en/2.4/suricata.html#troubleshooting-alerts

**Reply by s0meguy1** [reply]
Hi,

Yes I have run through those, just went through again to verify I didn't miss anything. To be clear, a default setup on any variation (bare metal ISO/github install from OS, VM) does not produce any alerts out of the box either from actual network traffic or from generating test data. From the first time SO walk-though, you should be able to see alerts from the generation of data, which doesn't work for me

**Reply by dougburks** [reply]
We do lots of automated and manual testing in many different variations and we're not experiencing this issue in our tests.

Can you provide more detail about what you found at each of the steps in the Troubleshooting Alerts section? That may help us pinpoint the problem faster.

- If you have metadata enabled, check to see if you have metadata for the connections. Depending on your configuration, this could be Suricata metadata or [Zeek](https://docs.securityonion.net/en/2.4/zeek.html#zeek) metadata.
- If you have metadata enabled but aren’t seeing any metadata, then something may be preventing the process from seeing the traffic. Check to see if you have any [BPF](https://docs.securityonion.net/en/2.4/bpf.html#bpf) configuration that may cause the process to ignore the traffic. If you’re sniffing traffic from the network, verify that the traffic is reaching the NIC using tcpdump. If importing a pcap file, verify that file contains the traffic you expect and that the Suricata process can read the file and any parent directories.
- Check your HOME_NET configuration to make sure it includes the networks that you’re watching traffic for.
- Check to see if you have a full NIDS ruleset with rules that should specifically alert on the traffic and that those rules are enabled.
- Check to see if you have any threshold or suppression configuration that might be preventing alerts.
- Check the Suricata log for additional clues.
- Check the [Elastic Agent](https://docs.securityonion.net/en/2.4/elastic-agent.html#elastic-agent), [Logstash](https://docs.securityonion.net/en/2.4/logstash.html#logstash), and [Elasticsearch](https://docs.securityonion.net/en/2.4/elasticsearch.html#elasticsearch) logs for any pipeline issues that may be preventing the alerts from being written to [Elasticsearch](https://docs.securityonion.net/en/2.4/elasticsearch.html#elasticsearch).
- Try installing a simple import node (perhaps in a VM) following the steps in the [First Time Users](https://docs.securityonion.net/en/2.4/first-time-users.html#first-time-users) section and see if you get alerts there. If so, compare the working system to the non-working system and determine where the differences are.

**Reply by s0meguy1** [reply]
1. If you have metadata enabled, check to see if you have metadata for the connections. Depending on your configuration, this could be Suricata metadata or [Zeek](https://docs.securityonion.net/en/2.4/zeek.html#zeek) metadata.
 * I have Suricata enabled and it seems to be collecting data:
 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/f33c6e6e-53b7-4886-a3bd-a63cf72fc8d5)

2.     If you have metadata enabled but aren’t seeing any metadata, then something may be preventing the process from seeing the traffic. Check to see if you have any [BPF](https://docs.securityonion.net/en/2.4/bpf.html#bpf) configuration that may cause the process to ignore the traffic. If you’re sniffing traffic from the network, verify that the traffic is reaching the NIC using tcpdump. If importing a pcap file, verify that file contains the traffic you expect and that the Suricata process can read the file and any parent directories.

bond0 is seeing traffic:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/3dca2daf-2926-4682-b1c6-327523b6a73d)


3.     Check your HOME_NET configuration to make sure it includes the networks that you’re watching traffic for.

HOME_NET set to default networks:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/db9cc8fd-c26a-4d47-b1d1-e37675e7aa4e)


5.     Check to see if you have a full NIDS ruleset with rules that should specifically alert on the traffic and that those rules are enabled.

I have the default ruleset in place, I haven't added anything as I cannot get any alerts or traffic to show in other dashboards (Hunt, kibana, etc)

6.     Check to see if you have any threshold or suppression configuration that might be preventing alerts.

I have not set anything as I am trying to troubleshoot seeing no traffic =.

7.     Check the Suricata log for additional clues.

 * No logs for 2 days, but these are the last logs:
```
 [11 - Suricata-Main] 2023-11-18 07:14:25 Info: logopenfile: tcp-data output device (regular) initialized: tcp-data.log
[11 - Suricata-Main] 2023-11-18 07:14:27 Info: detect: 1 rule files processed. 35310 rules successfully loaded, 0 rules failed
[11 - Suricata-Main] 2023-11-18 07:14:27 Info: threshold-config: Threshold config parsed: 0 rule(s) found
[11 - Suricata-Main] 2023-11-18 07:14:27 Info: detect: 35313 signatures processed. 1174 are IP-only rules, 5333 are inspecting packet payload, 28774 inspect application layer, 0 are decoder event only
[11 - Suricata-Main] 2023-11-18 07:14:30 Info: runmodes: bond0: creating 3 threads
[12 - W#01-bond0] 2023-11-18 07:14:30 Info: output-streaming: LogTcpDataLog is now set up
[13 - W#02-bond0] 2023-11-18 07:14:30 Info: output-streaming: LogTcpDataLog is now set up
[14 - W#03-bond0] 2023-11-18 07:14:30 Info: output-streaming: LogTcpDataLog is now set up
[11 - Suricata-Main] 2023-11-18 07:14:30 Info: unix-manager: unix socket '/var/run/suricata/suricata-command.socket'
[11 - Suricata-Main] 2023-11-18 07:14:30 Notice: threads: Threads created -> W: 3 FM: 1 FR: 1   Engine started.
```

8.     Check the [Elastic Agent](https://docs.securityonion.net/en/2.4/elastic-agent.html#elastic-agent), [Logstash](https://docs.securityonion.net/en/2.4/logstash.html#logstash), and [Elasticsearch](https://docs.securityonion.net/en/2.4/elasticsearch.html#elasticsearch) logs for any pipeline issues that may be preventing the alerts from being written to [Elasticsearch](https://docs.securityonion.net/en/2.4/elasticsearch.html#elasticsearch).

No errors in elasticsearch logs, the logs all look similar to this:
```
[2023-11-20T15:32:05,936][INFO ][org.elasticsearch.monitor.jvm.JvmGcMonitorService] [gc][young][443661][654] duration [996ms], collections [1]/[1.4s], total [996ms]/[2.3m], memory [810.9mb]->[307.6mb]/[10.5gb], all_pools {[young] [504mb]->[0b]/[0b]}{[old] [302.6mb]->[302.8mb]/[10.5gb]}{[survivor] [4.3mb]->[4.8mb]/[0b]}
[2023-11-20T15:32:05,936][WARN ][org.elasticsearch.monitor.jvm.JvmGcMonitorService] [gc][443661] overhead, spent [996ms] collecting in the last [1.4s]
[2023-11-20T15:43:13,475][INFO ][org.elasticsearch.xpack.security.action.role.TransportPutRoleAction] updated role [analyst]
[2023-11-20T15:43:15,083][INFO ][org.elasticsearch.xpack.security.action.role.TransportPutRoleAction] updated role [auditor]
[2023-11-20T15:43:15,507][INFO ][org.elasticsearch.xpack.security.action.role.TransportPutRoleAction] updated role [limited-analyst]
```


9.     Try installing a simple import node (perhaps in a VM) following the steps in the [First Time Users](https://docs.securityonion.net/en/2.4/first-time-users.html#first-time-users) section and see if you get alerts there. If so, compare the working system to the non-working system and determine where the differences are.

Did numerous installs as initial comment mentions

**Reply by dougburks** [reply]
> I have Suricata enabled and it seems to be collecting data:
>  [11 - Suricata-Main] 2023-11-18 07:14:25 Info: logopenfile: tcp-data output device (regular) initialized: tcp-data.log
[11 - Suricata-Main] 2023-11-18 07:14:27 Info: detect: 1 rule files processed. 35310 rules successfully loaded, 0 rules failed
[11 - Suricata-Main] 2023-11-18 07:14:27 Info: threshold-config: Threshold config parsed: 0 rule(s) found
[11 - Suricata-Main] 2023-11-18 07:14:27 Info: detect: 35313 signatures processed. 1174 are IP-only rules, 5333 are inspecting packet payload, 28774 inspect application layer, 0 are decoder event only
[11 - Suricata-Main] 2023-11-18 07:14:30 Info: runmodes: bond0: creating 3 threads
[12 - W#01-bond0] 2023-11-18 07:14:30 Info: output-streaming: LogTcpDataLog is now set up
[13 - W#02-bond0] 2023-11-18 07:14:30 Info: output-streaming: LogTcpDataLog is now set up
[14 - W#03-bond0] 2023-11-18 07:14:30 Info: output-streaming: LogTcpDataLog is now set up

The Suricata log screenshot and log output both look strange as we don't normally enable tcp-data.log. Have you made changes to the Suricata configuration?

> If you have metadata enabled, check to see if you have metadata for the connections. Depending on your configuration, this could be Suricata metadata or [Zeek](https://docs.securityonion.net/en/2.4/zeek.html#zeek) metadata.

When we refer to metadata here, we're talking about protocol metadata like DNS logs, HTTP logs, SSL/TLS logs, etc.:
https://docs.securityonion.net/en/2.4/suricata.html#metadata

By default, we use Zeek for this (although you can manually change this to Suricata). Do you have Zeek enabled? If so, do you have Zeek logs in /nsm/zeek/logs/current/ like dns.log, http.log, ssl.log?

> bond0 is seeing traffic:

Looking at the tcpdump screenshot, it looks like you're only seeing traffic from one internal IP address to another internal IP address. Are you seeing any traffic from an internal IP address to an external IP address? Have you double-checked your span port and cabling?

**Reply by s0meguy1** [reply]
>The Suricata log screenshot and log output both look strange as we don't normally enable tcp-data.log. Have you made changes to the Suricata configuration?

Yes, I enabled Suricata but I think thats the only mods I made

>By default, we use Zeek for this (although you can manually change this to Suricata). Do you have Zeek enabled? If so, do you have Zeek logs in /nsm/zeek/logs/current/ like dns.log, http.log, ssl.log?

I do have Zeek running, I didn't realize I hadn't turned it off completely. Here's a screenshot of that dir, which contains logs like you mentioned:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/f2c45716-d530-42b7-8a01-284f6d199f96)

>Looking at the tcpdump screenshot, it looks like you're only seeing traffic from one internal IP address to another internal IP address. Are you seeing any traffic from an internal IP address to an external IP address? Have you double-checked your span port and cabling?

Yes, for sure I have verified, its just a massive output and I selected the first traffic I saw for this post.

FYI - I just want to reiterate that on a fresh install (me touching nothing) I do not get anything. Right now I've poked with some things a bit, but I can confirm without touching anything and with confirmed traffic, I do not see alerts and am not able to send sample data

**Reply by dougburks** [reply]
> Yes, I enabled Suricata but I think thats the only mods I made

You shouldn't need to enable Suricata as it should be enabled by default. Also, you shouldn't need to make any changes to the Suricata configuration (like the tcp-data change noted above) for alerts to work.

> FYI - I just want to reiterate that on a fresh install (me touching nothing) I do not get anything. Right now I've poked with some things a bit, but I can confirm without touching anything and with confirmed traffic, I do not see alerts and am not able to send sample data

I understand and I want to help you figure this out. The best way to do so is to go back to first principles and eliminate as many variables as possible. Therefore, please download and verify the latest ISO image released yesterday (https://securityonion.net/download). Then perform a fresh installation following the steps at  https://blog.securityonion.net/2023/11/quick-malware-analysis-icedid-variant.html to do an IMPORT installation and then use so-import-pcap to import the specified pcap. After a minute or so, you should see alerts as shown in the second screenshot. If not, please provide more information about what is in /nsm/import after importing the pcap.

**Reply by s0meguy1** [reply]
Ok - I'll start fresh and report back. I'll be gone for the holidays so I more than likely will not be reporting back until next week. Thanks for the help so far!

**Reply by s0meguy1** [reply]
Heads up, I did get an error during install. I am not sure if I got the same error during the last install of 2.4.30. I've attached the error.log and sosetup.log, I'll leave it at that since I don't want to "contaminate" this clean install test. Standard Eval setup, I had the NUC management interface connected to internet during install, but nothing was connected to listening interface.

![Screenshot from 2023-11-21 16-19-17](https://github.com/Security-Onion-Solutions/securityonion/assets/25542926/e073e4e6-993b-4f32-916a-520fa25d0881)

[errors.log](https://github.com/Security-Onion-Solutions/securityonion/files/13431105/errors.log)
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/13431106/sosetup.log)

**Reply by dougburks** [reply]
That's definitely a problem. Looks like the Elastic Agent installation timed out and didn't install correctly. If Elastic Agent is not running correctly, then alerts will never be picked up.

The most likely culprit is disk I/O. Does this NUC have NVME, SSD, or rotational storage? If rotational, please try SSD or NVME and then perform a fresh installation to see if that helps. 

If you already have NVME or SSD, then are you still running Security Onion as a VM under Proxmox and do you have other VMs running? If so, then make sure that there are no other guests running and then perform a fresh installation to see if that helps. Also try a fresh installation of our latest ISO on bare metal (no Proxmox) to see if that helps.

**Reply by s0meguy1** [reply]
This is a bare metal (no VM) install on an NVME. You think this could be a bad drive? I will smack myself its its a bad drive...

**Reply by dougburks** [reply, marked as answer]
Interesting. It's possible that it's a bad drive, but can you share /opt/so/SO-Elastic-Agent_Installer.log so that we can see if there any other reasons why it may have timed out?

**Reply by s0meguy1** [reply]
Sure, I've attached.
[SO-Elastic-Agent_Installer.log](https://github.com/Security-Onion-Solutions/securityonion/files/13438958/SO-Elastic-Agent_Installer.log)

**Reply by defensivedepth** [reply]
@s0meguy1 

Let's double-check to confirm that the Elastic Agent was not installed:

`sudo elastic-agent status`

If that comes back with an error about elastic-agent not found, you can try manually installing the Elastic Agent:

`sudo salt-call state.apply elasticfleet.install_agent_grid`

**Reply by s0meguy1** [reply]
`sudo elastic-agent status`  failed as in not found, so I raned the salt-call command and got an error:

```
[ERROR   ] {'pid': 58546, 'retcode': 1, 'stdout': '\nInstallation initiated, view install log for further details.', 'stderr': '\n\nInstalling in non-interactive mode.: signal: killed'}
[INFO    ] Completed state [salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64] at time 13:39:48.840923 (duration_in_ms=185371.619)
local:
----------
          ID: run_installer
    Function: cmd.script
        Name: salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64
      Result: False
     Comment: Attempt 1: Returned a result of "False", with the following comment: "Command 'salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64' run"
              Command 'salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64' run
     Started: 13:36:03.219803
    Duration: 225596.244 ms
     Changes:
              ----------
              pid:
                  58546
              retcode:
                  1
              stderr:


                  Installing in non-interactive mode.: signal: killed
              stdout:

                  Installation initiated, view install log for further details.

Summary for local
------------
Succeeded: 0 (changed=1)
Failed:    1
```

I ran again and got the same error. Sorry that its cut off, I was running it in a screen session (bash from another machine, not screen in SO) because I was leaving my place and wanted to allow it to run. But it failed fairly quickly so I posted here

**Reply by s0meguy1** [reply]
Hi - I nabbed an SSD on black Friday, installed it and installed SO on it. Install went through with zero issues. SO is working as expected. I can't believe it was a bad drive. In any case, closing this

**Reply by defensivedepth** [reply]
Great to hear @s0meguy1
